fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java

```diff
@@ -520,10 +525,33 @@ public RecordMetaData getRecordMetaData() {
         return context.asyncToSync(FDBStoreTimer.Waits.WAIT_LOAD_META_DATA, getRecordMetaDataAsync(true));
     }
 
+    /**
+     * Save the record meta-data into the meta-data store.
+     *
+     * <p> If the file descriptor is missing a union descriptor, this method will automatically add one based on the non-{@code NESTED} record types. </p>
+     *
+     * @param fileDescriptor the file descriptor of the record meta-data
+     * @see MetaDataProtoEditor#addDefaultUnionIfMissing(Descriptors.FileDescriptor)
+     */
+    @API(API.Status.EXPERIMENTAL)
+    public void saveRecordMetaData(@Nonnull Descriptors.FileDescriptor fileDescriptor) {
+        saveRecordMetaData(RecordMetaData.build(MetaDataProtoEditor.addDefaultUnionIfMissing(fileDescriptor)));
```

Comment:
This doesn't take the old union from the store, which I believe means that it will assign sequential field numbers. These will then get overridden when `saveAndSetCurrent` merges with the persistent descriptor. Is that right? In particular, if a new type is earlier in the file, they'll all be off by one. Or am I missing something?

If I'm right, maybe a comment here would help us remember how this works.

And in any case, I think a test case like that is warranted.

Reply:
Yeah, that's right. Probably we should only allow this method if the meta-data store has no meta-data yet?

Reply:
I think restricting this to only saving if the meta-data hasn't been stored yet seems reasonable.

I guess the other problem case to consider would be what happens if someone calls `updateRecords` when the meta-data has been set like this. Like, in theory, you would want it to update the auto-generated union descriptor...I think. I suspect that it throws an error at the moment.

Reply:
Can we make it work? It seems like we've got most of the pieces needed. We need to make the new union descriptor out of the old union descriptor (if there is one) and the new local records descriptor.

Reply:
I'm okay with throwing errors, too, if it's tricky.

Reply:
My motivation is that I don't think a user when only a local records descriptor (no union) ever sees that they were doing a non-trivial evolution. From their point of view, these are the old records and these are the (legally evolved) new ones.

Reply:
So, does this become an alias (more or less) of `updateRecords`? I guess the difference is that this allows the meta-data to be absent at the beginning?

Reply:
Yeah, I guess so, now that I think about it.

If we decide to go with the error for now, then, thinking along those same lines, I believe we need a longer Javadoc on this method explaining that it's for getting the initial one saved and that you then need to use mutate (or whatever you need to use) to merge an updated local file descriptor with what it saved. The workflow isn't unbelievable onerous, but it's enough more complicated than just edit .proto, compile, save, repeat.

Reply:
Yeah, I think so. I think we should fix `updateRecords` here in this PR too. Then, `saveRecordMetaData` can just use `updateRecords`, when the meta-data already exists. I'll give it a try.

Reply:
> If we decide to go with the error for now ... 

I think we should be able to make it work. At least can give it a try.

Reply:
One more concern. Our initial intent was to limit the use of union-less meta-data to only `FDBMetaDataStore`. However, with the new changes, `RecrodMetaDataBuilder` now at least handles union-less local file descriptor. I.e., one could now start a record meta-data builder with a proto that has a union, but later drop the union in the local file. Could this make record manipulations unsafe? I cannot think of any problem at the moment, but it's worth noting.

Reply:
By record manipulations, do you mean, like, saving and loading records? Or do you mean schema evolutions?

I think it should be fine for saving/loading records as long as the synthetic union matches up correctly with the union in the stored meta-data. I think there's one case where an (implicitly) nested type might (surprise!) become a record type, but that's not an issue from using a local file descriptor, *per se*.

I think record evolution would still be safe. There's somewhat of a tradeoff here in that type name rather than union-descriptor position is now more important for determining whether the type in an older version is the same as the type in a newer one, but maybe that's okay.

Reply:
> By record manipulations, do you mean, like, saving and loading records? Or do you mean schema evolutions?

I meant `record meta-data manipulations`. thanks.